src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -80,25 +80,51 @@ Base.Array(A::LazyHcat) = stack(A.blocks)
 
 Base.adjoint(A::LazyHcat) = Adjoint(A)
 
-@views function Base.:*(Aadj::Adjoint{T,<:LazyHcat}, B::LazyHcat) where {T}
-    A = Aadj.parent
-    rows = size(A)[2]
-    cols = size(B)[2]
-    ret = similar(A.blocks[1], rows, cols)
-
-    orow = 0  # row offset
-    for blA in A.blocks
+# Computes A*B matrix product for LazyHcat type. Special case if product is assumed to be Hermitian
+@views function _mul(A::Adjoint{T,<:LazyHcat}, B::LazyHcat; hermitian=Val(false)) where {T}
+    Ap = A.parent
+    rows = size(Ap, 2)
+    cols = size(B, 2)
+    ret = similar(B.blocks[1], rows, cols)
+
+    if hermitian isa Val{true}
+        # Only popuplate the upper block diagonal in Hermitian case
         ocol = 0  # column offset
-        for blB in B.blocks
-            ret[orow .+ (1:size(blA, 2)), ocol .+ (1:size(blB, 2))] .= blA' * blB
+        for (ib, blB) in enumerate(B.blocks)
+            orow = 0  # row offset
+            for (ia, blA) in enumerate(Ap.blocks)
+                ib < ia && continue
```

Comment:
Isn't this (and the `Hermitian` wrapper) the only line that is not duplicate between the two versions of the `_mul` function ?

Reply:
Yes, that's the only difference